tests/rpc/gossamer_utils.go

```diff
@@ -0,0 +1,165 @@
+
+package rpc
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/rpc/modules"
+)
+
+var (
+	keyList = []string{"alice", "bob", "charlie", "dave", "eve", "fred", "george", "heather"}
+)
+
+// RunGossamer will start a gossamer instance and check if its online and returns CMD, otherwise return err
+func RunGossamer(t *testing.T, nodeNumb int, dataDir string) (*exec.Cmd, error) {
+
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+
+	gossamerCMD := filepath.Join(currentDir, "../..", "bin/gossamer")
+
+	//var cmd *exec.Cmd
+	//cmd = exec.Command(gossamerCMD, "init",
+	//	"--datadir", dataDir+strconv.Itoa(nodeNumb),
+	//	"--genesis", filepath.Join(currentDir, "../..", "node/gssmr/genesis.json"),
+	//	"--force",
+	//)
+	//
+	////add step for init
+	//t.Log("Going to init gossamer", "cmd", cmd)
+	//err = cmd.Start()
+	//if err != nil {
+	//	t.Error("Could not init gossamer", "err", err)
+	//	return nil, err
+	//}
+
+	//TODO: could we enable genesis file to be configured via args without init?
+	cmd := exec.Command(gossamerCMD, "--port", "700"+strconv.Itoa(nodeNumb),
+		"--key", keyList[nodeNumb],
+		"--datadir", dataDir+strconv.Itoa(nodeNumb),
+		"--rpchost", GOSSAMER_NODE_HOST,
+		"--rpcport", "854"+strconv.Itoa(nodeNumb),
+		"--rpcmods", "system,author,chain",
+		"--key", keyList[nodeNumb],
+		"--config", currentDir+"/config.toml",
+		"--roles", "4",
+		"--rpc",
+	)
+
+	t.Log("Going to execute gossamer", "cmd", cmd)
+	err = cmd.Start()
+	if err != nil {
+		t.Error("Could not execute gossamer cmd", "err", err)
+		return nil, err
+	}
+
+	t.Log("wait few secs for node to come up", "cmd.Process.Pid", cmd.Process.Pid)
+	var started bool
+
+	for i := 0; i < 10; i++ {
+		time.Sleep(3 * time.Second)
+		if err := CheckFunc(t, "http://"+GOSSAMER_NODE_HOST+":854"+strconv.Itoa(nodeNumb)); err == nil {
+			started = true
+			break
+		} else {
+			t.Log("Waiting for Gossamer to start", "err", err)
+		}
+	}
+	if started {
+		t.Log("Gossamer started :D", "cmd.Process.Pid", cmd.Process.Pid)
+	} else {
+		t.Fatal("Gossamer node never managed to start!")
+
+	}
+
+	return cmd, nil
+}
+
+// CheckFunc check if gossamer node is already started
+func CheckFunc(t *testing.T, gossamerHost string) error {
+	method := "system_health"
+
+	respBody := PostRPC(t, method, gossamerHost)
+
+	target := DecodeRPC(t, respBody, method)
+
+	if !target.(*modules.SystemHealthResponse).Health.ShouldHavePeers {
+		return fmt.Errorf("no peers")
+	}
+
+	//if we get here, we assume it worked :D
+
+	return nil
+}
+
+// KillGossamer kills a instance of gossamer
+func KillGossamer(t *testing.T, cmd *exec.Cmd) error {
+	err := cmd.Process.Kill()
+	if err != nil {
+		t.Log("failed to kill gossamer", "cmd", cmd)
+	}
+	return err
+}
+
+// bootstrap will spin gossamer nodes
+func Bootstrap(t *testing.T, pidList []*exec.Cmd) ([]*exec.Cmd, error) {
```

Comment:
since `t` is passed in here, do you need to return error or can you just do `t.Fatal()`?

Reply:
also, not sure about the name `Bootstrap`, maybe `StartNodes`?

Reply:
I like bootstrap better, but if you reeeeally think StartNodes is better then I can update.

Reply:
I would prefer something along the lines of `StartNodes` as well, it might be confusing since you are not testing the actual bootstrap configuration or bootstrap network methods.

Reply:
I dont think it makes any difference. The bootstrap is a term that comes as a standard term in the industry when you want to "bootstrap" the nodes. Because it includes also checks to make sure the nodes are started, calling it Start would miss the point.